# 2013-07-17 Stefan Teleman <stefan.teleman@oracle.com>
# 1. GCC did not pass the correct values-xpg?.c/values-X?.c when
# linking - based on language standard binding (-std=<xxx>.
#
# 2. C99 is not legal in C++98/C++03, but it is legal in C++11.
# We expose some of the C99 interfaces via different macros, such as
# _XPG5/_XOPEN_SOURCE=500 for C++98/C++03
#
# 3. Do not pass /usr/ccs/lib at link time (this directory is empty
# in S11 and S12
--- gcc-4.8.2/gcc/config/sol2.h	2013-01-10 12:38:27.000000000 -0800
+++ gcc-4.8.2/gcc/config/sol2.h	2013-10-17 11:15:49.102855258 -0700
@@ -89,8 +89,16 @@
 	   library.  */					\
 	if (c_dialect_cxx ())				\
 	  {						\
+	    if ((cxx_dialect == cxx98) || (cxx_dialect == cxx03)) \
+            {                                           \
+	      builtin_define ("__STDC_VERSION__=199409L"); \
+	      builtin_define ("_XOPEN_SOURCE=500");	\
+            }                                           \
+            else                                        \
+            {                                           \
 	    builtin_define ("__STDC_VERSION__=199901L");\
 	    builtin_define ("_XOPEN_SOURCE=600");	\
+            }                                           \
 	    builtin_define ("_LARGEFILE_SOURCE=1");	\
 	    builtin_define ("_LARGEFILE64_SOURCE=1");	\
 	    builtin_define ("__EXTENSIONS__");		\
